### PR TITLE
Convert release build to use generated matrix

### PR DESCRIPTION
**Pull request by TravisEz13**
## PR Summary

Convert release build to use generated matrix
This will make it easier to add LTS
This also has the benefit that each image is not individually retry-able. 

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
  - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Change is not breaking](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)
- [x] [Make sure all `Dockerfile`, `.sh`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
  - If the PR is work in progress, please add the prefix `WIP:` to the beginning of the title and remove the prefix when the PR is ready.
- **Adds a new image or tag**
  - [x] Not Applicable
- **OR**
  - [ ] Update [README.powershellcommunity.md](https://github.com/PowerShell/PowerShell-Docker/blob/master/assets/README.powershellcommunity.md)
  - [ ] Update [vsts-ci.yml](https://github.com/PowerShell/PowerShell-Docker/blob/master/vsts-ci.yml)

### CODE REVIEW

File: .vsts-ci/releasePhase.yml [hunk 1]
```diff
@@ -22,35 +29,56 @@ jobs:
...
+    condition: and( succeededOrFailed(), ne(variables['Channel'],''))
 
   - pwsh: ' az login --service-principal -u $(az_url) -p $(az_key) --tenant $(az_name)'
     displayName: 'az login'
+    condition: and( succeededOrFailed(), ne(variables['Channel'],''))
 
   - pwsh: |
       az account set --subscription $(StorageSubscriptionName)
-      $querystring = az storage account generate-sas --account-name $(StorageAccount) --services b --resource-types o --expiry ((get-date).AddDays(2).ToString("yyyy-MM-dd")) --permission r --https-only | convertfrom-json
+      $querystring = az storage account generate-sas --account-name $(StorageAccount) --services b --resource-types o --expiry ((get-date).AddDays(1).ToString("yyyy-MM-dd")) --permission r --https-only | convertfrom-json
```
Comment (adityapatwardhan) [inline .vsts-ci/releasePhase.yml]: any reason for changing from 2 to 1?

File: .vsts-ci/releaseStage.yml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        vmImage: ubuntu-16.04
+      displayName: Generate Matrix YAML ${{ parameters.channel }}
+      steps:
+      - pwsh: |
+          ./build.ps1 -GenerateMatrixJson -Channel ${{ parameters.channel }} -Verbose -Acr ${{ parameters.acr }} -OsFilter  ${{ parameters.osFilter }}
+        displayName: Generate Matrix YAML
+        condition: succeededOrFailed()
+        name: matrix
+      - pwsh: |
+          dir env:matrix_* | out-string -Width 1000
```
Comment (adityapatwardhan) [inline .vsts-ci/releaseStage.yml]: maybe also convertfrom-json to make it easier to review

**Comment by RDIL** [comment]
Looks good other then Aditya's feedback.

File: .vsts-ci/releasePhase.yml (hunk 1 above)
Comment (TravisEz13) [reply .vsts-ci/releasePhase.yml]: 2 days was for initial testing.  1 day is more than enough time.

File: .vsts-ci/releaseStage.yml (hunk 2 above)
Comment (TravisEz13) [reply .vsts-ci/releaseStage.yml]: The json is actually easier to read... When we get something like this

```

Name : json
json : @{a=}
```

**Pull request merged**